packages/hoppscotch-app/helpers/realtime/WSSessionAdapter.ts

```diff
@@ -0,0 +1,196 @@
+import { Ref } from "@nuxtjs/composition-api"
+import { NuxtApp } from "@nuxt/types/app"
+import { debounce } from "lodash"
+import { BehaviorSubject } from "rxjs"
+import { logHoppRequestRunToAnalytics } from "../fb/analytics"
+import { useStream } from "../utils/composables"
+import {
+  addWSLogLine,
+  addWSProtocol,
+  deleteWSProtocol,
+  setWSLog,
+  setWSProtocols,
+  setWSSocket,
+  updateWSProtocol,
+  WSLog$,
+  WSProtocols$,
+  WSSocket$,
+} from "~/newstore/WebSocketSession"
+
+export default class WSSessionAdapter {
+  error$: BehaviorSubject<string | null>
+  connecting$: BehaviorSubject<boolean>
+  connection$: BehaviorSubject<boolean>
+  urlValid$: BehaviorSubject<boolean>
+
+  socket: Ref<WebSocket>
```

Comment:
Instead of getting the Ref, try getting the BehaviourSubject itself here.

We are trying to remove all the Vue bits from here and move it out basically.